microcosm/hooks.py

```diff
@@ -42,15 +42,20 @@ class Bar:
 ON_RESOLVE = "_microcosm_on_resolve_"
 
 
-def _invoke_hook(hook_name, target):
+def _get_hook_name(hook_prefix, target_cls):
+    return f"{hook_prefix}_{target_cls.__name__}"
+
+
+def _invoke_hook(hook_prefix, target_component):
```

Comment:
I changed the name of the `target` argument here and in `_register_hook` to highlight the fact that hooks are registered on a class, and invoked on a component (i.e. an instance of that class).